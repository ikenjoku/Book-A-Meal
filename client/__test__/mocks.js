
export const meals = [
  {
    id: 0,
    name: 'Nigerian Jollof',
    description: 'Max Jollof rice, plantain, grilled turkey, bellefull Coke',
    price: 1100,
    imageurl: 'https://africa-publicmages/products/80/80418_1465475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 1,
    name: 'Rice and Beans',
    description: 'Rice, Beans, Plantain, Panla Sauce, Max Coke',
    price: 1500,
    imageurl: 'https://africa-g/images/products/80/80418_1465475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 2,
    name: 'Egusi Semo',
    description: 'Semolina, Egusi Soup, Meat, Extra',
    price: 1200,
    imageurl: 'https://africa-public.food.jcts/80/80418_1465475724_ma.jpg',
  },
  {
    id: 3,
    name: 'Green Calulu',
    description: 'Chicken salad with extra cabbage',
    price: 1250,
    imageurl: 'https://dynamic/production/ng/images/products/80/_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
  {
    id: 4,
    name: 'Onion Jappatti',
    description: 'Ingera palate with raw meat sauce',
    price: 2200,
    imageurl: 'https://dynamic/production/ng/images/produc475724_ma.jpg',
    createdAt: new Date(),
    updatedAt: new Date(),
  },
];

export const testMeal = {
  id: 3234,
  name: 'Strawberry flakes',
  description: 'Ingera palate with raw meat sauce',
  price: 2200,
  imageurl: 'https://dynamic/prodg/images/produc475724_ma.jpg',
  createdAt: new Date(),
  updatedAt: new Date(),
};

export const mockStoreData = {
  authReducer: {
    authLoading: true,
    user: {},
    isLoggedIn: false,
    error: null,
  },
  mealReducer: {
    meals: [
      {
        id: 3,
        name: 'Green Calulu',
        description: 'Chicken salad with extra cabbage',
        price: 1250,
        imageurl: 'https://dynamic/production/ng/images/products/80/_ma.jpg',
        createdAt: new Date(),
        updatedAt: new Date(),
      },
      {
        id: 4,
        name: 'Onion Jappatti',
        description: 'Ingera palate with raw meat sauce',
        price: 2200,
        imageurl: 'https://dynamic/production/ng/images/produc475724_ma.jpg',
        createdAt: new Date(),
        updatedAt: new Date(),
      },
    ],
    error: null,
    isLoading: false,

  },
  menuReducer: {
    selectedMenu: null,
    error: null,

  },
  setMenuReducer: {
    createdMenu: null,
    selectedDate: '',
    error: null,
    success: false,
    updatedMenu: null,
  },
  orderReducer: {
    orderedMeal: null,
    error: null,
    meals: [],
    orders: [],
    previousOrders: [],
    isLoadingOrders: false,
  },
};

export const createdMenu = {
  message: 'Menu for 2018-06-15 was successfully created',
  newMenu: {
    id: 24,
    date: '2018-06-15',
    updatedAt: '2018-07-02T17:37:46.023Z',
    createdAt: '2018-07-02T17:37:46.023Z',
  },
};

export const menus = [{
  id: 23,
  date: '2018-07-02',
  createdAt: '2018-07-01T23:56:23.508Z',
  updatedAt: '2018-07-01T23:56:23.508Z',
  Meals: [
    {
      id: 31,
      name: 'Beans and Bread',
      price: 800,
      description: 'Heavily loaded with smoked fish.',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529133645/bookameal/2018-06-16T07:20:43.822ZVegetablerice%2Bplantain%2Bgoatmeat%2Bcoke.jpg.jpg',
      MealMenu: {
        mealId: 31,
        menuId: 23,
        createdAt: '2018-07-01T23:56:28.012Z',
        updatedAt: '2018-07-01T23:56:28.012Z',
      },
    },
    {
      id: 34,
      name: 'Fish balls soup',
      price: 1200,
      description: 'Deliciously light soup with white fish balls',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529134583/bookameal/2018-06-16T07:36:22.300Zassortedmeat.jpg.jpg',
      MealMenu: {
        mealId: 34,
        menuId: 23,
        createdAt: '2018-07-01T23:56:32.313Z',
        updatedAt: '2018-07-01T23:56:32.313Z',
      },
    },
    {
      id: 30,
      name: 'Beef with veggie soup',
      price: 1150,
      description: 'Stir fried sliced beef with premium vegetables and spices',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529136037/bookameal/2018-06-16T08:00:34.302ZUgwu-Archi.jpg.jpg',
      MealMenu: {
        mealId: 30,
        menuId: 23,
        createdAt: '2018-07-01T23:56:36.920Z',
        updatedAt: '2018-07-01T23:56:36.920Z',
      },
    },
  ],
},
{
  id: 24,
  date: new Date().toISOString().substr(0, 10),
  createdAt: '2018-07-01T23:56:23.508Z',
  updatedAt: '2018-07-01T23:56:23.508Z',
  Meals: [
    {
      id: 31,
      name: 'Beans and Bread',
      price: 800,
      description: 'Heavily loaded with smoked fish.',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529133645/bookameal/2018-06-16T07:20:43.822ZVegetablerice%2Bplantain%2Bgoatmeat%2Bcoke.jpg.jpg',
      MealMenu: {
        mealId: 31,
        menuId: 23,
        createdAt: '2018-07-01T23:56:28.012Z',
        updatedAt: '2018-07-01T23:56:28.012Z',
      },
    },
    {
      id: 34,
      name: 'Fish balls soup',
      price: 1200,
      description: 'Deliciously light soup with white fish balls',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529134583/bookameal/2018-06-16T07:36:22.300Zassortedmeat.jpg.jpg',
      MealMenu: {
        mealId: 34,
        menuId: 23,
        createdAt: '2018-07-01T23:56:32.313Z',
        updatedAt: '2018-07-01T23:56:32.313Z',
      },
    },
    {
      id: 30,
      name: 'Beef with veggie soup',
      price: 1150,
      description: 'Stir fried sliced beef with premium vegetables and spices',
      imageurl: 'https://res.cloudinary.com/ikeenjoku/image/upload/v1529136037/bookameal/2018-06-16T08:00:34.302ZUgwu-Archi.jpg.jpg',
      MealMenu: {
        mealId: 30,
        menuId: 23,
        createdAt: '2018-07-01T23:56:36.920Z',
        updatedAt: '2018-07-01T23:56:36.920Z',
      },
    },
  ],
}];

export const orders = [
  {
    id: 2,
    date: '2018-05-16',
    amount: 1500,
    userId: 1,
    mealId: 2,
    status: 'cancelled',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 1,
      firstname: 'Ezinne',
      lastname: 'Njoku',
    },
    Meal: {
      id: 2,
      name: 'Rice and Beans',
      price: 1500,
    },
  },
  {
    id: 3,
    date: '2018-05-16',
    amount: 1200,
    userId: 2,
    mealId: 3,
    status: 'pending',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 2,
      firstname: 'Kelechi',
      lastname: 'Njoku',
    },
    Meal: {
      id: 3,
      name: 'Egusi Semo',
      price: 1200,
    },
  },
  {
    id: 4,
    date: '2018-05-16',
    amount: 1500,
    userId: 3,
    mealId: 3,
    status: 'delivered',
    createdAt: '2018-05-23T19:56:44.346Z',
    updatedAt: '2018-05-23T19:56:44.346Z',
    User: {
      id: 3,
      firstname: 'Nnamdi',
      lastname: 'James',
    },
    Meal: {
      id: 3,
      name: 'Egusi Semo',
      price: 1200,
    },
  },
];
