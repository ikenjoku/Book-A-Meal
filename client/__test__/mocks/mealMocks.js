export const getAllMealsMock = [
  {
    name: 'test meal',
    price: 1230,
    description: 'this is a very nice meal',
    imageurl: 'https://sdfsdjhbsn/sdf.asp',
  },
];
