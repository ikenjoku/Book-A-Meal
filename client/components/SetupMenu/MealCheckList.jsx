import React from 'react';
import PropTypes from 'prop-types';

const MealCheckList = ({ meals, handleSelect }) => {
  return meals.map(meal => (
    <div key={meal.id}>
      <label>
        <input
          type="checkbox"
          name="meals"
          value={meal.id}
          onClick={handleSelect}
          />
          <span>{meal.name}</span>
      </label>
    </div>
  ));
};

MealCheckList.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleSelect: PropTypes.func.isRequired,
};

export default MealCheckList;
