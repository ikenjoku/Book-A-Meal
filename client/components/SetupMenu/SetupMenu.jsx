import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import DatePicker from '../DatePicker';
import MenuCard from './MenuCard';
import { getAMenu, setSelectedDate } from '../../actions/setupMenuActions';

export class SetupMenu extends Component {
  componentDidMount() {
    const today = new Date().toISOString().substr(0, 10);
    this.props.getAMenu({ selectedDate: today });
    this.props.setSelectedDate({ selectedDate: today });
  }

  render() {
    return (
      <main id="setup-menu-container">
        <div className="cool-lg-text">
          <h2>Setup Menu</h2>
        </div>
        <DatePicker
          btnName="Find Menu"
          onSubmit={selectedDate => this.props.getAMenu(selectedDate)}
          onChange={selectedDate => this.props.setSelectedDate(selectedDate)}
        />
        <MenuCard />
      </main>
    );
  }
}

SetupMenu.propTypes = {
  getAMenu: PropTypes.func.isRequired,
  setSelectedDate: PropTypes.func.isRequired,
};

export default connect(null, { getAMenu, setSelectedDate })(SetupMenu);
