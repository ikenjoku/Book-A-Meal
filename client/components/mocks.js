

export const sampleOrders = {
  message: 'All orders retrieved successfully',
  orders: [
    {
      id: 2,
      date: '2018-05-16',
      amount: 1500,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 3,
      date: '2018-05-16',
      amount: 1200,
      userId: 2,
      mealId: 3,
      status: 'pending',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 4,
      date: '2018-05-16',
      amount: 1500,
      userId: 3,
      mealId: 3,
      status: 'delivered',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 3,
        firstname: 'Nnamdi',
        lastname: 'James',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 1,
      date: '2018-05-16',
      amount: 1100,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T20:38:56.387Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 5,
      date: '2018-05-23',
      amount: 1500,
      userId: 2,
      mealId: 2,
      status: 'pending',
      createdAt: '2018-05-23T23:55:05.091Z',
      updatedAt: '2018-05-23T23:55:05.091Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
    {
      id: 7,
      date: '2018-05-16',
      amount: 1200,
      userId: 2,
      mealId: 3,
      status: 'pending',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 2,
        firstname: 'Kelechi',
        lastname: 'Njoku',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 6,
      date: '2018-05-16',
      amount: 1500,
      userId: 3,
      mealId: 3,
      status: 'delivered',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T19:56:44.346Z',
      User: {
        id: 3,
        firstname: 'Nnamdi',
        lastname: 'James',
      },
      Meal: {
        id: 3,
        name: 'Egusi Semo',
        price: 1200,
      },
    },
    {
      id: 8,
      date: '2018-05-16',
      amount: 1100,
      userId: 1,
      mealId: 2,
      status: 'cancelled',
      createdAt: '2018-05-23T19:56:44.346Z',
      updatedAt: '2018-05-23T20:38:56.387Z',
      User: {
        id: 1,
        firstname: 'Ezinne',
        lastname: 'Njoku',
      },
      Meal: {
        id: 2,
        name: 'Rice and Beans',
        price: 1500,
      },
    },
  ],
};

