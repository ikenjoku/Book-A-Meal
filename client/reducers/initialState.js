export default {
  authReducer: {
    authLoading: true,
    user: {},
    isLoggedIn: false,
    error: null,
  },
  mealReducer: {
    meals: [],
    error: null,
    isLoading: true,

  },
  menuReducer: {
    selectedMenu: null,
    error: null,

  },
  setMenuReducer: {
    createdMenu: null,
    selectedDate: '',
    error: null,
    success: false,
    updatedMenu: null,
  },
  orderReducer: {
    orderedMeal: null,
    error: null,
    meals: [],
    orders: [],
    previousOrders: [],
    isLoadingOrders:false,
  },
};
