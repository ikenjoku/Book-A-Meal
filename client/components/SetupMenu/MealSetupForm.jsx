import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Pagination from 'react-js-pagination';
import { connect } from 'react-redux';
import { getAMeals } from '../../actions/setupMenuActions';
import MealCheckList from './MealCheckList';

export class MealSetupForm extends Component {
  state = {
    date: this.props.date || '',
    mealIds: [],
    activePage: 1,
  };

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.handleSubmit({
      date: this.state.date,
      mealIds: this.state.mealIds,
    });
  }

  handleSelect = (event) => {
    if (event.target.checked) {
      this.setState({
        mealIds: [
          ...this.state.mealIds,
          event.target.value,
        ],
      });
    } else {
      const currentMealIds = this.state.mealIds.filter(mealId =>
        mealId !== event.target.value);

      this.setState({ mealIds: currentMealIds });
    }
  }

  handlePageChange = (pageNumber) => {
    const limit = 5;
    this.props.getAMeals({ limit, page: pageNumber });
    this.setState(() => ({ activePage: pageNumber }));
  }

  render() {
    return (
      <form
        className="menu-setup-form"
        onSubmit={this.handleSubmit}
      >
        <div>
          <button className="calender-btn">
            <i className="far fa-calendar-alt fa-2x" />
          </button>
          <input
            readOnly
            name="date"
            type="date"
            value={this.state.date}
          />
        </div>
        <h3>Choose meals to add to menu</h3>
        <div className="meal-check-list">
          <MealCheckList
            meals={this.props.meals}
            handleSelect={this.handleSelect}
          />
        </div>
        <div className="center">
          <Pagination
            activePage={this.state.activePage}
            itemsCountPerPage={5}
            totalItemsCount={this.props.count}
            pageRangeDisplayed={5}
            onChange={this.handlePageChange}
          />
        </div>
        <div className="setup-btn-container">
          <button
            value="Submit"
            type="submit"
          >
            {this.props.action} Menu
          </button>
        </div>
      </form>
    );
  }
}

MealSetupForm.propTypes = {
  date: PropTypes.string.isRequired,
  getAMeals: PropTypes.func.isRequired,
  count: PropTypes.number.isRequired,
};

export const mapStateToProps = state => ({
  count: state.setupMenuReducer.count,
  pages: state.setupMenuReducer.pages,
});

MealSetupForm.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleSubmit: PropTypes.func.isRequired,
  action: PropTypes.string.isRequired,
};

export default connect(mapStateToProps, { getAMeals })(MealSetupForm);
