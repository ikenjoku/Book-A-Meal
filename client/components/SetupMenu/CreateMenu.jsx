import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import MealSetupForm from './MealSetupForm';
import { getAMeals, setAMenu } from '../../actions/setupMenuActions';

class CreateMenu extends Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.getAMeals({limit: 20, page: 1});
  }

  handleSubmit = (values) => {
    this.props.setAMenu(values)
    this.props.closeModal();
  }

  render() {
    const {
      meals, isCreating, isFetching,
    } = this.props;

    return (
      <div className='setup-form'>
        <h2>Create a menu</h2>
        {isFetching
          ?
          <h3>Fetching meals...</h3>
          :
          <MealSetupForm
            handleSubmit={this.handleSubmit}
            meals={meals}
            action="Create"
            isSubmitting={isCreating}
            date={this.props.selectedDate}
          />}
      </div>
    );
  }
}

CreateMenu.propTypes = {
  getAMeals: PropTypes.func.isRequired,
  setAMenu: PropTypes.func.isRequired,
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  isFetching: PropTypes.bool.isRequired,
  isCreating: PropTypes.bool.isRequired,
  closeModal: PropTypes.func.isRequired,
  selectedDate: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  meals: state.setupMenuReducer.meals,
  isFetching: state.setupMenuReducer.isFetching,
  isCreating: state.setupMenuReducer.isCreating,
  selectedDate: state.setupMenuReducer.selectedDate,
});

export default connect(mapStateToProps, { getAMeals, setAMenu })(CreateMenu);