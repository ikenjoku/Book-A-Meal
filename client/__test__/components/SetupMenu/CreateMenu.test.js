import React from 'react';
import { shallow } from 'enzyme';
import { CreateMenu ,mapStateToProps } from '../../../components/SetupMenu/CreateMenu';
import initialState from '../../../reducers/initialState'
import { meals } from '../../mocks';

describe('CreateMenu Component', () => {
  let wrapper, closeModal, setAMenu, getAMeals, isFetching, isCreating, selectedDate;
  setAMenu = jest.fn();
  getAMeals = jest.fn();
  closeModal = jest.fn();
  isFetching = false;
  isCreating = false;
  selectedDate = '2018-09-23';

  test('should render CreateMenu correctly', () => {
    wrapper = shallow(
      <CreateMenu
      setAMenu={setAMenu}
      getAMeals={getAMeals}
      closeModal={closeModal}
      meals={meals}
      isFetching={isFetching}
      isCreating={isCreating}
      />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should render Loading when fetching meals', () => {
    wrapper = shallow(
      <CreateMenu
      setAMenu={setAMenu}
      getAMeals={getAMeals}
      closeModal={closeModal}
      meals={meals}
      isFetching={true}
      isCreating={isCreating}
      />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should handleSubmit when called with mealIds', () => {
    wrapper = shallow(
      <CreateMenu
      setAMenu={setAMenu}
      getAMeals={getAMeals}
      closeModal={closeModal}
      meals={meals}
      isFetching={false}
      isCreating={isCreating}
      />);

    const handleSubmitSpy = jest.spyOn(
      wrapper.instance(), 'handleSubmit'
    );

    wrapper.instance().handleSubmit([1, 2, 3]);
    expect(handleSubmitSpy).toHaveBeenCalledTimes(1);
  });

  test('should map state to props', () => {
    const ownProps = { 
      meals:{meals},
      isCreating: {isCreating}
    };
    const tree = mapStateToProps(initialState, ownProps);
    expect(tree).toMatchSnapshot();
  });
});
