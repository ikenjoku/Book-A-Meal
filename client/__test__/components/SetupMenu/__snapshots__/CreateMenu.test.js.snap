// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateMenu Component should map state to props 1`] = `
Object {
  "isCreating": false,
  "isFetching": false,
  "meals": Array [],
  "selectedDate": "",
}
`;

exports[`CreateMenu Component should render CreateMenu correctly 1`] = `
<div
  className="setup-form"
>
  <h2>
    Create a menu
  </h2>
  <Connect(MealSetupForm)
    action="Create"
    handleSubmit={[Function]}
    isSubmitting={false}
    meals={
      Array [
        Object {
          "createdAt": "2018-08-27T19:48:26.757Z",
          "description": "Max Jollof rice, plantain, grilled turkey, bellefull Coke",
          "id": 0,
          "imageurl": "https://africa-publicmages/products/80/80418_1465475724_ma.jpg",
          "name": "Nigerian Jollof",
          "price": 1100,
          "updatedAt": "2018-08-27T19:48:26.757Z",
        },
        Object {
          "createdAt": "2018-08-27T19:48:26.757Z",
          "description": "Rice, Beans, Plantain, Panla Sauce, Max Coke",
          "id": 1,
          "imageurl": "https://africa-g/images/products/80/80418_1465475724_ma.jpg",
          "name": "Rice and Beans",
          "price": 1500,
          "updatedAt": "2018-08-27T19:48:26.757Z",
        },
        Object {
          "description": "Semolina, Egusi Soup, Meat, Extra",
          "id": 2,
          "imageurl": "https://africa-public.food.jcts/80/80418_1465475724_ma.jpg",
          "name": "Egusi Semo",
          "price": 1200,
        },
        Object {
          "createdAt": "2018-08-27T19:48:26.757Z",
          "description": "Chicken salad with extra cabbage",
          "id": 3,
          "imageurl": "https://dynamic/production/ng/images/products/80/_ma.jpg",
          "name": "Green Calulu",
          "price": 1250,
          "updatedAt": "2018-08-27T19:48:26.757Z",
        },
        Object {
          "createdAt": "2018-08-27T19:48:26.757Z",
          "description": "Ingera palate with raw meat sauce",
          "id": 4,
          "imageurl": "https://dynamic/production/ng/images/produc475724_ma.jpg",
          "name": "Onion Jappatti",
          "price": 2200,
          "updatedAt": "2018-08-27T19:48:26.757Z",
        },
      ]
    }
  />
</div>
`;

exports[`CreateMenu Component should render Loading when fetching meals 1`] = `
<div
  className="setup-form"
>
  <h2>
    Create a menu
  </h2>
  <h3>
    Fetching meals...
  </h3>
</div>
`;
